src/Strategy/Ruby/Gemspec.hs

```diff
@@ -0,0 +1,108 @@
+module Strategy.Ruby.Gemspec (
+  rubyString,
+  parseRubyAssignment,
+  Assignment (..),
+  readAssignments,
+  parseRubyArray,
+  parseRubyWordsArray,
+  rubyLicenseValuesP,
+) where
+
+import Control.Applicative ((<|>))
+import Data.Char (isSpace)
+import Data.Functor (($>))
+import Data.List.Extra (singleton)
+import Data.String.Conversion (toText)
+import Data.Text (Text)
+import Data.Void (Void)
+import Text.Megaparsec (MonadParsec (eof), Parsec, anySingle, anySingleBut, between, choice, lookAhead, many, optional, sepBy, skipManyTill, takeWhile1P, takeWhileP, try)
+import Text.Megaparsec.Char (char, space, space1, string)
+
+type Parser = Parsec Void Text
+
+-- |Given a single start delimiter, return start/end delimiters
+selectDelim :: Char -> (Char, Char)
+selectDelim = \case
+  '{' -> ('{', '}')
+  '<' -> ('<', '>')
+  '(' -> ('(', ')')
+  '[' -> ('[', ']')
+  c -> (c, c)
+
+rubyString :: Parser Text
```

Comment:
[nit] Would like to see comment with example text it ought to be parsing (few cases). Here is good [example](https://github.com/fossas/fossa-cli/blob/master/src/App/Fossa/VSI/DynLinked/Internal/Binary.hs#L108). 

I'm guilty of not doing this too, but given this parser seems bit involved, example case and edge case illustrated would help in reviewing. 

This applies to other applicable parser as well in this PR.

Reply:
Hi Megh, I've updated with text around each of the parsers. I hope they make it a little more clear what they do